### PR TITLE
refactor(config): nest age and vehicles

### PR DESCRIPTION
**Pull request by kaligrafy**
nest in projectConfig with shared asset validations

Group age thresholds under projectConfig.age and vehicle limits under projectConfig.vehicles, keeping maxHouseholdSize at the root (default 18). Add householdAssetCountValidation for car, bicycle, and two-wheel counts, using household.size × maxPerPerson with a maxHouseholdSize fallback when size is unknown. Wire participant validations and household audits to the new config, keep audit messages static, and update demo configs, mocks, and parametric tests.

## Summary by CodeRabbit

- **New Features**
  - Added configurable validation for household cars, bicycles, and two-wheelers based on household size and project limits.
  - Added localized English and French messages for household and vehicle validation errors.

- **Improvements**
  - Consolidated age, vehicle, and household-size settings into clearer configuration sections.
  - Audit checks now use configured limits instead of fixed thresholds.
  - Updated validation and warning messages for greater clarity.

- **Tests**
  - Expanded coverage for asset limits, age boundaries, configuration defaults, and validation edge cases.

### CODE REVIEW

File: packages/evolution-common/src/config/__tests__/project.config.test.ts [hunk 1]
```diff
@@ -11,12 +11,20 @@ import { ISODateTimeStringWithTimezoneOffset } from '../../utils/DateTimeUtils';
 test('Expected default', () => {
     expect(projectConfig).toEqual(expect.objectContaining({
         region: 'CA',
-        selfResponseMinimumAge: 14,
-        drivingLicenseAge: 16,
-        workingAge: 15,
-        schoolMandatoryAge: 15,
-        maxPersonAge: 125,
-        addAuditWarningVeryOldAge: undefined,
+        age: expect.objectContaining({
```
Comment (tahini) [inline packages/evolution-common/src/config/__tests__/project.config.test.ts]: ça ne devrait pas être expect.objectContaining, c'est ça that's it that's all, non? objectContaining, c'est si on s'attend à plus de champs

File: packages/evolution-common/src/config/__tests__/project.config.test.ts [hunk 2]
```diff
@@ -35,10 +43,21 @@ test('Expected default', () => {
 test('set project configuration', () => {
     const configToSet = {
         region: 'FR',
-        selfResponseMinimumAge: 18,
-        drivingLicenseAge: 16,
-        workingAge: 16,
-        schoolMandatoryAge: 17,
+        age: {
+            selfResponseMinimumAge: 18,
+            interviewableAge: 5,
```
Comment (tahini) [inline packages/evolution-common/src/config/__tests__/project.config.test.ts:54]: Ici, il ne faudrait pas tous les mettre, pour s'assurer que les valeurs par défaut sont bien conservées pour les valeurs manquantes. À moins qu'il y ait déjà un autre test qui fait ça?

File: packages/evolution-common/src/config/project.config.ts [hunk 3]
```diff
@@ -57,49 +90,10 @@ export type EvolutionProjectConfiguration = {
...
-     */
-    maxPersonAge: number;
-    /**
-     * Age from which a person age triggers a reviewer warning audit (inclusive).
-     * Applies up to {@link maxPersonAge}. When undefined, no age warning audit is raised.
-     */
-    addAuditWarningVeryOldAge?: number;
+    /** Maximum household size for widgets and validations. Defaults to 18. */
+    maxHouseholdSize: number;
+    age: EvolutionAgeConfiguration;
```
Comment (tahini) [inline packages/evolution-common/src/config/project.config.ts]: ce champ ne devrait-il pas être plutôt au pluriel? Et une petite typedoc au-dessus ne ferait pas de tort

**Review by tahini — COMMENTED** [review]
Il devrait y avoir au moins 2 commits dans cette PR. Juste renommer les âges, c'est un commit tout seul, sans autre changement. Renommer les véhicules aussi, à la limite, ça peut aller dans le même commit. Mais les ajouts de validations, ça va à part, les 2 ne sont pas liés.

File: packages/evolution-common/src/config/__tests__/project.config.test.ts (hunk 2 above)
Comment (kaligrafy) [reply packages/evolution-common/src/config/__tests__/project.config.test.ts:54]: ajouté tous

File: packages/evolution-common/src/services/questionnaire/sections/segments/widgetSegmentModePre.ts [hunk 4]
```diff
@@ -33,7 +33,7 @@ const canPersonDriveCar: WidgetConditional = (interview) => {
         person && person.drivingLicenseOwnership !== undefined ? person.drivingLicenseOwnership : 'dontKnow';
     if (
         drivingLicenseOwnership === 'dontKnow' &&
-        ((person.age && person.age > config.drivingLicenseAge) || !person.age)
+        ((person.age && person.age > config.ages.drivingLicenseAge) || !person.age)
```
[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-common/src/services/questionnaire/sections/segments/widgetSegmentModePre.ts]: fixed

File: packages/evolution-common/src/config/__tests__/project.config.test.ts (hunk 1 above)
Comment (kaligrafy) [reply packages/evolution-common/src/config/__tests__/project.config.test.ts]: fixed

File: packages/evolution-common/src/config/project.config.ts (hunk 3 above)
Comment (kaligrafy) [reply packages/evolution-common/src/config/project.config.ts]: fixed

File: packages/evolution-common/src/services/widgets/validations/householdAssetCountValidation.ts [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+            },
+            {
+                validation: Number(value) < 0,
+                errorMessage: messages.minZero
+            },
+            {
+                validation: Number(value) > maxTotal,
+                errorMessage: messages.overMax(maxPerPerson)
+            }
+        ];
```
[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-common/src/services/widgets/validations/householdAssetCountValidation.ts:100]: Please create a new issue for this, out of scope of this PR. Do not assign anybody yet. Put label Medium Priority

File: packages/evolution-backend/src/services/auth/__tests__/authByFieldLogin.test.ts [hunk 6]
```diff
@@ -72,7 +72,13 @@ jest.mock('../../../models/interviewsPreFill.db.queries', () => ({
 
 // Mock the project config to set the postalCodeRegion
 jest.mock('evolution-common/lib/config/project.config', () => ({
-    postalCodeRegion: 'canada'  // Default for tests
+    postalCodeRegion: 'canada', // Default for tests
+    maxHouseholdSize: 18,
```
Comment (tahini) [inline packages/evolution-backend/src/services/auth/__tests__/authByFieldLogin.test.ts]: Avoir cette config n'est probablement pas nécessaire pour un test de authByFieldLogin...

File: packages/evolution-common/src/services/widgets/validations/__tests__/householdAssetCountValidation.test.ts [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    validationFn: typeof carNumberValidation,
+    value: unknown,
+    interview: any
+) => validationFn(value, undefined, interview, 'household.carNumber').findIndex((rule) => rule.validation);
+
+describe('household asset count validation', () => {
+    describe('getMaxHouseholdAssetCount', () => {
+        it.each([
+            [4, 3, 12],
+            [5, 5, 25]
```
Comment (tahini) [inline packages/evolution-common/src/services/widgets/validations/__tests__/householdAssetCountValidation.test.ts]: Il faudrait tester avec des 0 aussi et voir ce que ça donne avec des nombres négatifs

File: packages/evolution-common/src/services/widgets/validations/householdAssetCountValidation.ts [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+    };
+};
+
+/** @see {@link ValidationFunction} */
+export const carNumberValidation = createHouseholdAssetCountValidation(
+    projectConfig.vehicles.maxCarsPerHouseholdMember,
+    {
+        invalid: {
+            fr: 'Le nombre de véhicules est invalide.',
```
Comment (tahini) [inline packages/evolution-common/src/services/widgets/validations/householdAssetCountValidation.ts]: On doit éviter ceci dans le code, ce n'est pas traductible par la suite, préférer des strings dans les fichiers locales, avec une fonction `t`.

File: packages/evolution-common/src/services/widgets/validations/householdAssetCountValidation.ts [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+ * Build a widget validation for a household asset count (cars, bicycles, two-wheelers).
+ * The upper bound is {@link getMaxHouseholdAssetCount} using {@link getEffectiveHouseholdSizeForAssetCount}.
+ *
+ * Widgets should define a `conditional` that hides the asset input until `household.size`
+ * is completed and valid (integer ≥ 1), so participants answer household size first.
+ *
+ * @param maxPerPerson maximum count allowed per household member
+ * @param messages localized validation messages
+ */
+export const createHouseholdAssetCountValidation = (
```
Comment (tahini) [inline packages/evolution-common/src/services/widgets/validations/householdAssetCountValidation.ts]: Beaucoup de fonctions dans ce fichier semblent des détails d'implémentation, comme celle-ci. A-t-elle vraiment besoin d'être exportée? Si c'est uniquement pour les tests unitaires, la réponse est non, il faut tester l'API des fonctions de ce fichier, pas leur implémentation.

File: packages/evolution-common/src/services/widgets/validations/validations.ts [hunk 10]
```diff
@@ -85,10 +84,10 @@ export const householdSizeValidation: ValidationFunction = (value) => {
             }
         },
         {
-            validation: Number(value) > 18,
+            validation: Number(value) > projectConfig.maxHouseholdSize,
             errorMessage: {
-                fr: 'La taille du ménage doit être au maximum 18.',
-                en: 'Household size must be at most 18.'
+                fr: `La taille du ménage doit être au maximum ${projectConfig.maxHouseholdSize}.`,
```
Comment (tahini) [inline packages/evolution-common/src/services/widgets/validations/validations.ts]: Ici aussi, idéalement, il faudrait mettre ça dans un fichier locale. Cette approche est très legacy.

File: packages/evolution-common/src/services/questionnaire/sections/segments/__tests__/widgetSegmentModePre.test.ts [hunk 11]
```diff
@@ -229,34 +229,32 @@ describe('Mode choices conditionals', () => {
...
-        // Prepare test data with active person/journey/trip
-        const interview = _cloneDeep(interviewAttributesForTestCases);
-        interview.response._activePersonId = 'personId3';
-        interview.response._activeJourneyId = 'journeyId3';
-        interview.response._activeTripId = 'tripId1P3';
-        interview.response.household!.persons!.personId3!.age = 15;
+    test.each([
+        ['person age is unknown', undefined, true],
+        ['person is too young', 15, false],
+        ['person is at driving license age', 16, true]
```
Comment (tahini) [inline packages/evolution-common/src/services/questionnaire/sections/segments/__tests__/widgetSegmentModePre.test.ts]: need a person with age older too

**Review by tahini — COMMENTED** [review]
Quelques considérations. Aussi, le premier commit est un breaking change qui demandera de changer la config dans les enquêtes. SVP ajouter une ligne dans le changelog expliquant le changement.

File: packages/evolution-common/src/services/widgets/validations/validations.ts (hunk 10 above)
Comment (kaligrafy) [reply packages/evolution-common/src/services/widgets/validations/validations.ts]: Corrigé ceux qui sont dans cette PR. Les autres legacy: #1786

File: packages/evolution-common/src/services/questionnaire/sections/segments/__tests__/widgetSegmentModePre.test.ts (hunk 11 above)
Comment (kaligrafy) [reply packages/evolution-common/src/services/questionnaire/sections/segments/__tests__/widgetSegmentModePre.test.ts]: added

File: packages/evolution-common/src/services/widgets/validations/householdAssetCountValidation.ts (hunk 9 above)
Comment (kaligrafy) [reply packages/evolution-common/src/services/widgets/validations/householdAssetCountValidation.ts]: removed

File: packages/evolution-common/src/services/widgets/validations/householdAssetCountValidation.ts (hunk 8 above)
Comment (kaligrafy) [reply packages/evolution-common/src/services/widgets/validations/householdAssetCountValidation.ts]: corrigé

File: packages/evolution-common/src/services/widgets/validations/__tests__/householdAssetCountValidation.test.ts (hunk 7 above)
Comment (kaligrafy) [reply packages/evolution-common/src/services/widgets/validations/__tests__/householdAssetCountValidation.test.ts]: added

File: packages/evolution-backend/src/services/auth/__tests__/authByFieldLogin.test.ts (hunk 6 above)
Comment (kaligrafy) [reply packages/evolution-backend/src/services/auth/__tests__/authByFieldLogin.test.ts]: removed

**Review by tahini — APPROVED** [review]
